src/Framework/Framework/Resources/Scripts/typings/knockout/knockout.dotvvm.d.ts

```diff
@@ -7,3 +7,21 @@ interface KnockoutDelaySync {
     resume(): void;
     run(action: () => void): void;
 }
+
+interface KnockoutUtils {
+    cloneNodes<T extends Node>(nodesArray: T[], shouldCleanNodes?: boolean): T[];
+}
+
+interface KnockoutObservableStatic {
+    <T>(value: T, validator: (newValue: T) => void): KnockoutObservable<T>;
+}
+interface KnockoutObservableArrayStatic {
+    <T>(value: T[] | null, validator: (newValue: T[]) => void): KnockoutObservableArray<T>;
+}
+
+interface KnockoutObservable<T> {
+    readonly state?: T;
+    patchState?(patch: any): void;
+    setState?(newState: any): void;
+    readonly updater?: UpdateDispatcher<T>;
```

Comment:
Using `T` in this definition is technically wrong, since T will have inner properties wrapped in knockout observables, while `state` and `updater` will be fully unwrapped.

We already have `DotvvmObservable<T>` which handles this in the "opposite" way - it wraps all properties of `T` in observables. 

I think it would be better to adjust methods like `ko.dataFor` and `ko.contextFor` to return object with `DotvvmObservable`s.